libvast/vast/concept/parseable/vast/expression.hpp

```diff
@@ -49,11 +49,12 @@ struct predicate_parser : parser<predicate_parser> {
     using parsers::chr;
     using namespace parser_literals;
     auto id = +(alnum | chr{'_'} | chr{'-'});
+    auto key_char = alnum | chr{'_'} | chr{'-'} | chr{':'};
     // A key cannot start with ':', othwise it would be interpreted as a type
     // extractor.
-    auto key = !':'_p >> (+(alnum | chr{'_'} | chr{'-'} | chr{':'}) % '.');
+    auto key = !(':'_p | '-') >> (+key_char % '.');
     auto operand
-      = parsers::data ->* to_data_operand
+      = (parsers::data >> !key_char) ->* to_data_operand
```

Comment:
Is end-of-input (`eoi`) also "not a key char"?

That is, does it work when the operand is either LHS or RHS?

Reply:
`eoi` is not a _char_. If `data` consumed the last character, `key_char` will fail no matter what it is composed of.